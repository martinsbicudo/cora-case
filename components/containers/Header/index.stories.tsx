import { StoryObj, Meta } from '@storybook/react'

import Header from '.'

export default {
  title: 'Components / Header',
  component: Header,
} as Meta

export const Default: StoryObj = {}
