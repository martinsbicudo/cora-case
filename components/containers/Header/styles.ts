import styled from 'styled-components'

export const Header = styled.header`
  padding: 24px 0;
  border-bottom: 1px solid ${({ theme }) => theme.colors['gray-3']};
`

export const HeaderTitle = styled.h1`
  font-weight: 900;
  font-size: 2.2rem;
  line-height: 3.2rem;
  color: ${({ theme }) => theme.colors['off-black']};
`
