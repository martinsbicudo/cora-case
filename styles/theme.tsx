const theme = {
  colors: {
    primary: '#FE3E6D',
    secondary: '#1A93DA',
    'soft-secondary': '#EAF7FF',
    'off-black': '#3B3B3B',
    'gray-1': '#6B7076',
    'gray-2': '#C7CBCF',
    'gray-3': '#DEE4E9',
    'gray-4': '#F0F4F8',
    white: '#FFFFFF',
  },
}

export default theme
