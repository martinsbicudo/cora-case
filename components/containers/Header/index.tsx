import { Container } from '@Commons'

import * as Styled from './styles'

const Header = () => {
  return (
    <Styled.Header>
      <Container>
        <Styled.HeaderTitle>Extrato</Styled.HeaderTitle>
      </Container>
    </Styled.Header>
  )
}

export default Header
