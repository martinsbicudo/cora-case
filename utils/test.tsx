import type { ReactElement, ReactNode } from 'react'

import { ThemeProvider } from 'styled-components'

import theme from '@/styles/theme'
import { render, RenderOptions } from '@testing-library/react'

const Providers = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'
export { customRender as render }
