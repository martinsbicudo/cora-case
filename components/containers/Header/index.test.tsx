import theme from '@/styles/theme'
import { render, screen } from '@/test/helpers'

import Header from '.'

describe('Containers / Header', () => {
  const renderComponent = () => render(<Header />)

  it('should render component', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })
})
