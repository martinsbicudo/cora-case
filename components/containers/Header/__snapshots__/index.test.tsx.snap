// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers / Header should render component 1`] = `
<div>
  <header
    class="sc-jrcTuL AEVqD"
  >
    <div
      class="sc-gswNZR iKhUPL"
    >
      <h1
        class="sc-kDvujY kndZqW"
      >
        Extrato
      </h1>
    </div>
  </header>
</div>
`;
